src/azure/cli/commands/arm.py

```diff
@@ -160,11 +160,22 @@ def required_values_validator(namespace):
 
 APPLICATION.register(APPLICATION.COMMAND_TABLE_LOADED, add_id_parameters)
 
-def register_generic_update(name, getter, setter, factory=None, setter_arg_name='parameters', #pylint:disable=too-many-arguments
-                            simple_output_query=None, custom_handlers=None):
+def _get_child_property(parent, collection_name, item_name):
+    items = getattr(parent, collection_name)
+    result = next((x for x in items if x.name.lower() == item_name.lower()), None)
+    if not result:
+        raise CLIError("Property '{}' does not exist".format(item_name))
+    else:
+        return result
+
+def cli_generic_update_command(name, getter, setter, factory=None, setter_arg_name='parameters', #pylint:disable=too-many-arguments
+                               simple_output_query=None, child_arg_type=None,
```

Comment:
I went with 'child_collection_prop_name' since it's the property name of the child collection on the client.

---
In reply to: [75968958](https://github.com/Azure/azure-cli/pull/735#discussion_r75968958) [](ancestors = 75968958)